subscriptions/subscriptions-impl/src/main/java/com/duckduckgo/subscriptions/impl/SubscriptionsManager.kt

```diff
@@ -382,6 +388,11 @@ class RealSubscriptionsManager @Inject constructor(
         return !userHadFreeTrial && privacyProFeature.get().privacyProFreeTrial().isEnabled() && freeTrialProductsAvailableInGooglePlay
     }
 
+    override suspend fun isSwitchPlanAvailable(): Boolean {
+        val hasActiveSubscription = authRepository.getSubscription()?.isActive() ?: false
+        return hasActiveSubscription && privacyProFeature.get().supportsSwitchSubscription().isEnabled()
```

Comment:
Accessing FF value might involve blocking IO operation, so it's better to wrap it with `withContext(dispatchers.io) {}` just in case.